tools/plugins/package/src/main/java/net/corda/cli/plugins/packaging/aws/kms/KmsKeyStore.java

```
@@ -0,0 +1,151 @@
+import net.corda.cli.plugins.packaging.aws.kms.rsa.KmsRSAKeyFactory;
+import org.jetbrains.annotations.NotNull;
+import software.amazon.awssdk.services.kms.KmsClient;
+import software.amazon.awssdk.services.kms.model.AliasListEntry;
+import software.amazon.awssdk.services.kms.model.DescribeKeyResponse;
+import software.amazon.awssdk.services.kms.model.ListAliasesRequest;
+import software.amazon.awssdk.services.kms.model.ListAliasesResponse;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.security.Key;
+import java.security.KeyStoreException;
+import java.security.KeyStoreSpi;
+import java.security.NoSuchAlgorithmException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.util.Collections;
+import java.util.Date;
+import java.util.Enumeration;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class KmsKeyStore extends KeyStoreSpi {
+
+    @NotNull
+    private final KmsClient kmsClient;
+
+    public KmsKeyStore(KmsClient kmsClient) {
+        Objects.requireNonNull(kmsClient);
+        this.kmsClient = kmsClient;
+    }
+
+    @Override
+    public Enumeration<String> engineAliases() {
+        return Collections.enumeration(getAliases().stream().map(AliasListEntry::aliasName).collect(Collectors.toSet()));
+    }
+
+    @Override
+    public boolean engineContainsAlias(String alias) {
+        String prefixedAlias = getPrefixedAlias(alias);
+        return getAliases().stream().filter(e -> e.aliasName().equals(prefixedAlias)).findAny().isPresent();
+    }
+
+    @Override
+    public int engineSize() {
+        return getAliases().size();
+    }
+
+    @Override
+    public Key engineGetKey(String alias, char[] chars) throws NoSuchAlgorithmException, UnrecoverableKeyException {
```

It looks like this is not used, which sounds correct as we never want to extract the key from the keystore. So maybe we should throw a NotSupported exception here instead of keeping the implementation. I realise that this would divert the code from the original, but it would prevent somebody accidentally making a change that would involve extracting the key.

---

If I understand the code correctly, it never retrieves the key from the keystore/KMS. It creates a `Kms[EC/RSA]PrivateKey` object that holds only `keyId`.